system/DataCaster/DataCaster.php

```diff
@@ -59,18 +65,20 @@ final class DataCaster
     ];
 
     /**
-     * @param array<string, class-string>|null $castHandlers Custom convert handlers
-     * @param array<string, string>|null       $types        [field => type]
-     * @param object|null                      $helper       Helper object.
-     * @param bool                             $strict       Strict mode? Set to false for casts for Entity.
+     * @param TCastHandlers|null         $castHandlers Custom convert handlers
+     * @param array<string, string>|null $types        [field => type]
+     * @param object|null                $helper       Helper object.
+     * @param bool                       $strict       Strict mode? Set to false for casts for Entity.
      */
     public function __construct(
         ?array $castHandlers = null,
         ?array $types = null,
         private readonly ?object $helper = null,
         private readonly bool $strict = true,
     ) {
-        $this->castHandlers = array_merge($this->castHandlers, $castHandlers);
+        if ($castHandlers !== null && $castHandlers !== []) {
+            $this->castHandlers = array_merge($this->castHandlers, $castHandlers);
+        }
```

Comment:
Can this be just `$this->castHandlers = array_merge($this->castHandlers, $castHandlers ?? []);`?